htdocs/core/modules/modTakePos.class.php

```diff
@@ -266,9 +267,45 @@ public function __construct($db)
 	 */
 	public function init($options = '')
 	{
-		global $conf, $db;
+		global $conf, $db, $langs, $user;
+		$langs->load("cashdesk");
 
 		dolibarr_set_const($db, "TAKEPOS_PRINT_METHOD", "browser", 'chaine', 0, '', $conf->entity);
+		
+		//Default customer for Point of sale
+		if (empty(getDolGlobalInt('CASHDESK_ID_THIRDPARTY1'))) {
+			$societe = new Societe($db);
+			$societe->nom = $langs->trans("DefaultPOSThirdLabel");
+			$societe->client = 1;
+			$result = $societe->create($user);
+			if ($result > 0) {
+				dolibarr_set_const($db, "CASHDESK_ID_THIRDPARTY1", $result, 'chaine', 0, '', $conf->entity);
+			} else {
+				setEventMessages($societe->error, $societe->errors, 'errors');
+			}
+		}
+		
+		//Create category if not exists
+		$categories = new Categorie($this->db);
+		$cate_arbo = $categories->get_full_arbo('product', 0, 1);
+		if (is_array($cate_arbo)) {
+			if (!count($cate_arbo)) {
+				$category = new Categorie($db);
+				$category->label = $langs->trans("DefaultPOSCatLabel");
+				$category->type = Categorie::TYPE_PRODUCT;
+				$result = $category->create($user);
+				if ($result > 0) {
```

Comment:
I understand addition of a Default Category and Thirdparty, but what is the goald of this product ?

Reply:
For TakePOS to work you must have created at least one product. If there are no products created, TakePOS says it needs at least one product to work. A generic product may be interesting to be able to use TakePOS without configuration. Then if the user wants to change the name or create more products, they are welcome.